src/sentry/tasks/assemble.py

```diff
@@ -367,21 +367,18 @@ def _bind_or_create_artifact_bundle(
         return existing_artifact_bundle, False
 
 
-def _mark_bundles_that_need_indexing(
-    associated_bundles: List[ArtifactBundle], release: str, dist: str
-):
-    # We update all bundles that have the "does not need indexing" state to "needs indexing". Since we might have
-    # concurrent tasks that modify the "indexing_state" concurrently, we will issue an update statement that checks
-    # if the "indexing_state" is equal to "does not need indexing" and only in that case we update it. This is done
-    # in order to implement atomic compare_and_set semantics because we will perform the update when locking the row.
+def _get_bundles_that_need_indexing(associated_bundles: List[ArtifactBundle]):
+    # We get all the bundles that are not indexed. This implementation is not safe for concurrency, since the values
+    # might change afterward, however, we do use a Redis cache to prevent duplicate work. That cache is also not 100%
+    # concurrency safe, but it will help for most cases.
```

Comment:
as we are not concurrency-safe anyway, you might as well just filter on `bundle.indexing_state` directly, instead of doing a roundtrip through the database. You already have fetched all the `associated_bundles` from the DB just a couple of calls before.

Reply:
Will do